internal/plan/plan.go

```diff
@@ -211,6 +211,59 @@ func (s *Service) Equal(other *Service) bool {
 	return reflect.DeepEqual(s, other)
 }
 
+// ParseCommand returns a service command as two stream of strings.
+// The base command is returned as a stream and the default arguments
+// in [ ... ] group is returned as another stream.
+func (s *Service) ParseCommand() (base, extra []string, err error) {
+	args, err := shlex.Split(s.Command)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var inBrackets, gotBrackets bool
+
+	for idx, arg := range args {
+		if inBrackets {
+			if arg == "[" {
+				return nil, nil, fmt.Errorf("cannot nest [ ... ] groups")
+			}
+			if arg == "]" {
+				inBrackets = false
+				continue
+			}
+			extra = append(extra, arg)
+			continue
+		}
+		if gotBrackets {
+			return nil, nil, fmt.Errorf("cannot have any args after [ ... ] group")
+		}
+		if arg == "[" {
+			if idx == 0 {
+				return nil, nil, fmt.Errorf("command cannot be started with default arguments")
```

Comment:
`cannot start command with [ ... ] group`, as above and below.

Reply:
Done.